lib/iris/fileformats/netcdf.py

```diff
@@ -780,6 +784,13 @@ def write(self, cube, local_keys=None, unlimited_dimensions=None,
                     'cf_patch')
                 warnings.warn(msg)
 
+        # Perform any user registered callback function.
+        # As the NetCDF saver commits to disk as it processes the cube and
+        # its metadata, it is not possible to skip or ignore the derived
+        # NetCDF representation of a cube as with other file format savers.
```

Comment:
Totally agree. The **loader** callback is also somewhat odd for `NetCDF`. For the loader I opted to just pass in the `netCDF4.variable` of the associated cube.

I've never, ever used a callback on netCDF for loading. For me, the nature/architecture of the file format just doesn't fit the callback scheme.